cpp/src/io/orc/reader_impl.cu

```diff
@@ -222,8 +222,8 @@ class metadata {
       CUDF_EXPECTS(row_count >= 0, "Invalid row count");
       CUDF_EXPECTS(row_start <= get_total_rows(), "Invalid row start");
 
-      int stripe_skip_rows = 0;
-      for (int i = 0, count = 0; i < (int)ff.stripes.size(); ++i) {
+      size_type stripe_skip_rows = 0;
+      for (size_type i = 0, count = 0; i < (size_type)ff.stripes.size(); ++i) {
```

Comment:
Prefer C++ explicit cases to C-style casts.
```suggestion
      for (size_type i = 0, count = 0; i < static_cast<size_type>(ff.stripes.size()); ++i) {
```

Reply:
Changed i to size_t instead (avoids the cast altogether ~ personally, I prefer C-style casts on trivial types for readability, though I understand that goes against libcudf convention)

Reply:
Actually, while making these changes, it became apparent that the use of size_type severely limits our ability to extract rows from large ORC files. Start row would ideally need to be size_t in the API (though using rowgroup-based on stripe-based selection should bypass these restrictions).

Reply:
So made a bunch of changes in that area (albeit all outside the initial scope of this PR), in order to attempt loading fewer than 2B rows from a large orc/parquet file that may contain a number of rows well outside the 31-bit size_type limit.

Reply:
> Changed i to size_t instead (avoids the cast altogether ~ personally, I prefer C-style casts on trivial types for readability, though I understand that goes against libcudf convention)

One thing to consider is that although C-style casts are "easier to read" in that the take less characters - they don't express intent and you can't actually know what the programmer meant. You will definitely run into a case where you think it is "trivial" but you end up doing a `const_cast` instead of a `static_cast`, or worse you end up doing a `reinterpret_cast`.

Item 2 of "More Effective C++" is Prefer C++ style casts.